Yafc.UI/Core/DrawingSurface.cs

```diff
@@ -41,10 +41,14 @@ public abstract class DrawingSurface(float pixelsPerUnit) : IDisposable {
 
         public abstract Window window { get; }
 
+#pragma warning disable CA1816 
+        // "CA1816: Dispose methods should call SuppressFinalize." 
+        // Suppressing this inspection because it wants to add GC.SuppressFinalize, but do we really want to suppress finalization? I'm not sure, but I don't think we do.
```

Comment:
Yes you want to call `SuppressFinalize`, unless your code has some unmanaged resources that are not cleaned up in the dispose method, but are cleaned up in your finalizer.

Now considering the finalizer looks like this:
```C#
        ~MemoryDrawingSurface() {
            Dispose();
        }
```

I think it's pretty safe to use `supresFinalize` inside the dispose method. (Unless i missed some other inheritance of this class that has a different finalizer)

Reply:
Looking at this a bit more:

A better question is, why is `Dispose` implemented in `DrawingSurface` but the finalizer 2 inherence level deeper in `MemoryDrawingSurface`?

A more proper implementation of the dispose pattern would be like this: https://github.com/Dorus/yafc-ce/commit/65f36aa258170119a0522544de9706f54ab10980

Reply:
Thank you for your explanation, very educational (for me at least)

Reply:
Thank you for explanation! I vote for keeping this part of the PR as-is, and then test the new implementation of Dispose in another PR.

Reply:
That would be the safe route yes.
@Dorus let's make a new PR with your improvements and test separately?

Reply:
Yeah, i've spend several hours reading up on garbage collection, finalizers and the dispose pattern before it started making sense to me. 🤣

One thing you should not do is assume that it's all some kind of magic. It's all fairly straightforwards and what you see is what you get kind of thing.

One wrong assumption i had is that GC would call dispose. It doesn't. Instead it will call the finalizer, but only do so when it finally (lol) decides to dispose of an object. The garbage collector works in the background at its own pace, and will not run too often, or for small objects. It uses internal heuristics to decide what to try to collect (cue in zero gen, first gen, second gen GC steps, where it checks if an object is unreachable early, and then bag them all together into increasingly less inspected groups). Once an object is long living, GC will only visit it and check if it's still reachable if the bucket it's in is large, and memory pressure is high. But unmanaged resources throw a wrench into this process, a bytes large object might hold megabytes of unmanaged resources, but the GC only sees those bytes. Unmanaged can also hold other resources like file handles, window objects. network connection etc.

Thus the `IDisposable` pattern is introduced for developers to manually call cleanup methods for unmanaged resources. This is aided in the language with `using`  keywords etc. This is also why dispose suggest you `null` large fields: You will be calling dispose before the object itself is unreachable. Adding a finalizer to call Dispose is just a fallback in case you forget to dispose manually, to make a memory leak less likely. But there is no guarantee that the GC process will ever call this, or call it in a timely matter.